test/ListM.lean

```diff
@@ -22,13 +22,16 @@ open Lean
   let x := ((ListM.fix F 10).force).run []
   guard $ x = some ([10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
 
-#check (rfl : ((ListM.fix F 10).takeAsList 4).run [] = some ([10, 9, 8, 7], [7, 8, 9, 10]))
-#check (rfl :
-  (((ListM.fix F 10).map fun n => n*n).takeAsList 3).run [] = some ([100, 81, 64], [8, 9, 10]))
+example : ((ListM.fix F 10).takeAsList 4).run [] = some ([10, 9, 8, 7], [8, 9, 10]) := by
```

Comment:
There are two changes here:
 1. `rfl` no longer unfolds `ListM` operations because they're partial and/or opaque.  (But the `rfl` was `unsafe` anyhow.)
 2. I optimized `ListM.fix` so that it requires fewer invocations of `F` to produce the same number of elements.  (That's why the second part of the pair---which stores each invocation of `F`---got shorter.)